include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -573,6 +573,44 @@ class __future : private std::tuple<_Args...>
     }
 };
 
+// A contract for a future class for reduce: <execution policy, sycl::event, USM host memory for the reduced value>
+template <typename _ExecutionPolicy, typename _Event, typename _Res>
+class __reduce_future
+{
+    _ExecutionPolicy __my_exec;
+    _Event __my_event;
+    _Res __my_res;
+
+  public:
+    __reduce_future(_ExecutionPolicy __exec, _Event __e, _Res __res)
```

Comment:
Usually you construct the instance of this class like
```__reduce_future(::std::forward<_ExecutionPolicy>(__exec), ...```
but this declaration mean that move semantic here really isn't possible, I think.